src/sql/src/plan/hir.rs

```diff
@@ -3423,6 +3522,58 @@ impl HirScalarExpr {
             }
         })
     }
+
+    /// Attempts to simplify this expression of [`ScalarType::Int64`] to a literal Int64 and
+    /// returns it as an i64.
+    ///
+    /// Returns `PlanError::ConstantExpressionSimplificationFailed` if
+    /// - it's not a constant expression (as determined by `is_constant`)
+    /// - evaluates to null
+    /// - an EvalError occurs during evaluation (e.g., a cast fails)
+    ///
+    /// # Panics
+    ///
+    /// Panics if this expression does not have type [`ScalarType::Int64`].
+    pub fn try_into_literal_int64(self) -> Result<i64, PlanError> {
+        // TODO: add the `is_constant` check also to all the other into_literal_... (by adding it to
+        // `simplify_to_literal`), but those should be just soft_asserts at first that it doesn't
+        // actually happen that it's weaker than `reduce`, and then add them for real after 1 week.
+        // (Without the is_constant check, lower_uncorrelated's preconditions spill out to be
+        // preconditions also of all the other into_literal_... functions.)
+        if !self.is_constant() {
+            return Err(PlanError::ConstantExpressionSimplificationFailed(format!(
+                "Expected a constant expression, got {}",
+                self
+            )));
+        }
+        self.clone()
+            .simplify_to_literal_with_result()
+            .and_then(|row| {
+                let datum = row.unpack_first();
+                if datum.is_null() {
+                    Err(PlanError::ConstantExpressionSimplificationFailed(format!(
+                        "Expected an expression that evaluates to a non-null value, got {}",
+                        self
+                    )))
+                } else {
+                    Ok(datum.unwrap_int64().cast_into())
```

Comment:
I think the `cast_into` is unnecessary, `unwrap_int64` returns `i64`.

Reply:
Indeed, thanks! This was leftover from when this used to be unwrapping a UInt64 and then returning a usize.